flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/AsynchronousException.java

```diff
@@ -38,6 +38,10 @@ public AsynchronousException(String message, Throwable cause) {
 
 	@Override
 	public String toString() {
-		return "AsynchronousException{" + getCause() + "}";
+		if (getMessage() != null) {
+			return "AsynchronousException{" + getMessage() + ", caused by " + getCause() + "}";
+		} else {
+			return "AsynchronousException{" + getCause() + "}";
+		}
```

Comment:
I'd be in favour of simply removing the `toString` method.

Reply:
Done.

This is a more intrusive change that may break more tests as it  impacts the route where the message was null.

Any objection to me changing the tests to assert on the default AsyncException.toString() rendering?